src/App.jsx

```diff
@@ -19,6 +19,7 @@ function App() {
         <Routes>
           <Route path="/" element={<ArticleList topic={topic} />} />
           <Route path="/articles" element={<ArticleList topic={topic} />} />
```

Comment:
By putting topic in state, after choosing a topic I now cannot navigate back to all articles. I'd suggest working with queries rather than topic state and getting your topic from the url.